src/OrbitGl/GlCanvas.cpp

```diff
@@ -241,29 +195,35 @@ void GlCanvas::LeftUp() {
 
 void GlCanvas::LeftDoubleClick() {
   double_clicking_ = true;
+  SetPickingMode(PickingMode::kClick);
   RequestRedraw();
 }
 
 void GlCanvas::RightDown(int x, int y) {
-  float world_x, world_y;
-  ScreenToWorld(x, y, world_x, world_y);
+  screen_click_ = Vec2i(x, y);
+  world_click_pos_ = viewport_.ScreenToWorldPos(screen_click_);
 
-  select_start_ = select_stop_ = Vec2(world_x, world_y);
+  select_start_ = select_stop_ = world_click_pos_;
   is_selecting_ = true;
+  SetPickingMode(PickingMode::kClick);
 
   Orbit_ImGui_MouseButtonCallback(imgui_context_, 1, true);
   RequestRedraw();
 }
 
 bool GlCanvas::RightUp() {
   Orbit_ImGui_MouseButtonCallback(imgui_context_, 1, false);
-  is_selecting_ = true;
+  is_selecting_ = false;
+  select_start_ = select_stop_;
```

Comment:
This line makes the green selection rectangle disappear on "right-up" which is a change in behavior. I think we should keep the previous behavior.

Reply:
Thanks for catching this, fixed now!